service/src/main/kotlin/fi/espoo/evaka/messaging/MessageQueries.kt

```diff
@@ -1219,6 +1221,35 @@ fun Database.Read.getMessageAccountsForRecipients(
     val childRecipients = groupedRecipients[MessageRecipientType.CHILD]?.map { it.id } ?: listOf()
     val citizenRecipients =
         groupedRecipients[MessageRecipientType.CITIZEN]?.map { it.id } ?: listOf()
+
+    val filterPredicates =
+        PredicateSql.allNotNull(
+            if (filters?.yearsOfBirth?.isNotEmpty() == true) {
+                PredicateSql {
+                    where(
+                        "date_part('year', p.date_of_birth) = ANY('{${filters.yearsOfBirth.joinToString(",")}}')"
```

Comment:
Use `${bind()}` instead of manually formatting parameters into the raw SQL string. In this case there's no possibility of an SQL injection vulnerability, but manual hard-coding is error-prone and defeats prepared statement caching. In *some rare cases* this is a good approach, but even then I'd personally want to see a comment saying it's intentional